lib-find-meal/src/main/java/com/kinandcarta/lib/find/meal/view/FindMealFragment.kt

```diff
@@ -73,141 +42,61 @@ class FindMealFragment : Fragment() {
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        super.onViewCreated(view, savedInstanceState)
-
-        progressBarView = view.findViewById(R.id.progressBar)
-        mealsRecyclerView = view.findViewById(R.id.rvMeals)
-        swipeRefreshLayout = view.findViewById(R.id.swipeContainer)
-
-        viewModel = ViewModelProvider(this).get(FindMealViewModel::class.java)
-
-        fusedLocationClient = FusedLocationProviderClient(this.requireActivity())
-        mealsRecyclerView.adapter = mealsAdapter
-        mealsRecyclerView.layoutManager = LinearLayoutManager(context)
-
-        swipeRefreshLayout.setOnRefreshListener {
-            swipeRefreshLayout.isRefreshing = false
-            getLastLocation()
-        }
-
-        getLastLocation()
-    }
-
-    private fun confirmPermissions(context: Context) {
-        if (ContextCompat.checkSelfPermission(
-                context,
-                Manifest.permission.ACCESS_FINE_LOCATION
-            ) !== PackageManager.PERMISSION_GRANTED
-        ) {
-            requestFineLocationPermission(LOCATION_PERMISSION_REQUEST_CODE)
-        }
-    }
-
-    fun requestFineLocationPermission(requestCode: Int = 0) {
-        requestPermissions(arrayOf(Manifest.permission.ACCESS_FINE_LOCATION), requestCode)
+        setupUIWithView(view)
+        viewModel.state.observe(viewLifecycleOwner, ::onStateChanged)
     }
 
+    @SuppressLint("MissingPermission")
     override fun onRequestPermissionsResult(
         requestCode: Int,
         permissions: Array<out String>,
         grantResults: IntArray
     ) {
+        val permissionResultParser = PermissionResultParser()
         if (permissionResultParser.isFineLocationPermissionsGranted(permissions, grantResults)) {
-            getLastLocation()
+            viewModel.startUpdatingLocation()
```

Comment:
the `startUpdatingLocation` function requires location permissions, but since we're doing this in this callback function we can just suppress the linting since we're confident that they've given us permissions